pkg/backup/backup.go

```diff
@@ -154,30 +162,38 @@ func (b *Backup) Run() {
 		var ackchan chan backupNowAck
 		select {
 		case <-time.After(interval):
-		case ackchan = <-b.backupNow:
+		case ackchan = <-bc.backupNow:
 			logrus.Info("received a backup request")
 		}
 
-		rev, err := b.saveSnap(lastSnapRev)
+		rev, bs, err := bc.backupManager.SaveSnap(lastSnapRev)
 		if err != nil {
 			logrus.Errorf("failed to save snapshot: %v", err)
 		}
 		lastSnapRev = rev
 
+		if bs != nil {
+			bc.recentBackupsStatus = append(bc.recentBackupsStatus, *bs)
+			if len(bc.recentBackupsStatus) > maxRecentBackupStatusCount {
+				bc.recentBackupsStatus = bc.recentBackupsStatus[1:]
+			}
+		}
+
 		if ackchan != nil {
 			ack := backupNowAck{err: err}
 			if err == nil {
-				ack.status = b.getLatestBackupStatus()
+				ack.status = bc.recentBackupsStatus[len(bc.recentBackupsStatus)-1]
 			}
 			ackchan <- ack
 		}
 	}
 }
 
-func (b *Backup) saveSnap(lastSnapRev int64) (int64, error) {
-	podList, err := b.kclient.Core().Pods(b.namespace).List(k8sutil.ClusterListOpt(b.clusterName))
+// SaveSnap saves the snapshot if snapshot's revision is greater than the given lastSnapRev.
```

Comment:
explain the return value.

Reply:
also does backupstate contain rev already?

Reply:
will do what you suggested.